snuba/query/snql/parser.py

```diff
@@ -734,6 +735,44 @@ def parse_snql_query_initial(
     return parsed
 
 
+def _qualify_columns(query: Union[CompositeQuery[QueryEntity], LogicalQuery]) -> None:
+    """
+    All columns in a join query should be qualified with the entity alias, e.g. e.event_id
+    Take those aliases and put them in the table name. This has to be done in a post
+    process since we need to have all the aliases from the join clause.
+    """
+
+    from_clause = query.get_from_clause()
+    if isinstance(query, LogicalQuery) or not isinstance(from_clause, JoinClause):
+        print("REJECTED")
+        return  # We don't qualify columns that have a single source
+
+    # mypy makes this difficult to do non-recursively
```

Comment:
Is there any specific drawback on the recursive approach? It seems to me the most natural way to navigate through hierarchical structures.

Reply:
No, it's a personal thing. I thought I could do it in a single loop without needing a new function definition. Since it's already implemented it doesn't really matter.